pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeSystem.java

```diff
@@ -50,4 +53,11 @@ public int getMaxNumericScale() {
   public int getMaxNumericPrecision() {
     return MAX_DECIMAL_PRECISION_DIGIT;
   }
+
+  @Override
+  public RelDataType deriveAvgAggType(RelDataTypeFactory typeFactory,
```

Comment:
Great catch! You will need to make some changes to the `WindowFunctionsPlans.json` to pick up the casting in the plans (I've used `AVG` quite extensively in that file)

Reply:
Resolved all test issues.